zipline/protocol.py

```diff
@@ -194,6 +200,87 @@ def current_portfolio_weights(self):
         return position_values / self.portfolio_value
 
 
+class AlgorithmPortfolio(Portfolio):
+
+    def __init__(self, data_portal, benchmark=None):
+        super(AlgorithmPortfolio, self).__init__()
+        self.data_portal = data_portal
+        self.benchmark = benchmark
+        self.current_date = None
+
+    @property
+    def expected_shortfall(self):
+        """
+        Function for computing expected shortfall (also known as CVaR, or
+        Conditional Value at Risk) for the portfolio according to the assets
+        currently held and their respective weight in the portfolio.
+
+        This function requires a data portal in order to retrieve price
+        histories of the assets in the portfolio.
+        """
+        data_portal = self.data_portal
+        current_date = self.current_date
+
+        # If we do not even have a year's worth of data to look back on
+        # then the expected shortfall calculation will not be reliable, so
+        # just return NaN. If we only have between one and two years of
+        # data just use what is available.
+        num_days_of_data = data_portal.trading_calendar.session_distance(
+            self.start_date, current_date,
```

Comment:
Like we discussed this is incorrect, but I think what I want here is the start date of _all_ data. Either that, or later on when I do a history call I'll have to catch `HistoryWindowStartsBeforeData` errors.